Compilation error, "Cannot deduce type of copy" / -enzyme-print-activity

Hi and many thanks for your amazing work! I'm trying to apply Enzyme to a particle physics code and stumbled across the unexpected behavior detailed below.

**Steps to reproduce**: Compile the following code, also uploaded to https://fwd.gymni.ch/zN6ioW:
```c++
#include <vector>

struct A {
  std::vector<int> v;
  A(){
    v.resize(2);
  }
};

double f(double x){
  A* a = new A;
  return 0;
}

extern double __enzyme_autodiff(void*, double);

int main(int argc, char* argv[]) {
  __enzyme_autodiff((void*)f, 3.14);
} 
```
Compiled (A) with the Enzyme explorer and (B) on my local system (Enzyme built from bad7df07dd7657c0c6884667d62ad1d9bcfd1d16, Ubuntu clang version 14.0.0-1ubuntu1.1 and  LLVM version 14.0.0) via
```bash
clang++-14 source.cpp -S -emit-llvm -o input.ll -O2 -fno-vectorize -fno-slp-vectorize -fno-unroll-loops
opt-14 --enable-new-pm=0 input.ll -load=[...]/LLVMEnzyme-14.so -enzyme -o output.ll -S -enzyme-print-activity 
```

**Expected behaviour**: Compilation and optimization passes run successfully.

**Observed behaviour**: (A) The Enzyme Explorer shows a compiler error
```
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/vector:66:
/usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/stl_uninitialized.h:1011:2: error: Enzyme: Cannot deduce type of copy   call void @llvm.memmove.p0i8.p0i8.i64(i8* align 4 %11, i8* align 4 %12, i64 %13, i1 false) #18, !dbg !539<analysis>i64 4: {[-1]:Integer}, intvals: {4,}i64 0: {[-1]:Anything}, [....]</analysis>
        __builtin_memmove(__result, __first, __count * sizeof(_Tp));
```
(B) `opt` outputs the following error message: [error.txt](https://github.com/EnzymeAD/Enzyme/files/13817582/error.txt). Without `-enzyme-print-activity`, it runs fine.

--- a/enzyme/Enzyme/ActivityAnalysis.cpp
+++ b/enzyme/Enzyme/ActivityAnalysis.cpp
@@ -2192,7 +2192,7 @@ bool ActivityAnalyzer::isConstantValue(TypeResults const &TR, Value *Val) {
       else
         llvm::errs() << potentiallyActiveStore;
       llvm::errs() << " potentialStore=";
-      if (potentiallyActiveStore)
+      if (potentialStore)
         llvm::errs() << *potentialStore;
       else
         llvm::errs() << potentialStore;

